src/sentry/workflow_engine/migrations/0051_migrate_remaining_issue_alerts.py

```diff
@@ -2240,12 +2241,12 @@ def migrate_issue_alert(rule: Any) -> None:
             )
             sentry_sdk.capture_exception(e)
 
-    migrated_issue_alerts = AlertRuleWorkflow.objects.filter(rule_id__isnull=False).values_list(
-        "rule_id", flat=True
+    migrated_issue_alerts = AlertRuleWorkflow.objects.filter(
+        rule_id=OuterRef("id"), rule_id__isnull=False
     )
 
     for rule in RangeQuerySetWrapperWithProgressBarApprox(
-        Rule.objects.filter(id__gt=progress_id).exclude(id__in=migrated_issue_alerts)
+        Rule.objects.filter(~Exists(migrated_issue_alerts), id__gt=progress_id)
```

Comment:
```suggestion
        Rule.objects.exclude(Exists(migrated_issue_alerts), id__gt=progress_id)
```
the `filter` and `~` are kinda like a boolean saying `is_disabled = True`, by changing filter to `exclude` we can just use the exists.

Reply:
oops

Reply:
nbd, sorry i forgot to hit  enter on the review 😅

Reply:
i have to refactor anyway because the query is still too slow 😭